static/app/utils/replays/resourceFrame.tsx

```diff
@@ -53,3 +55,100 @@ export function getResponseBodySize(frame: SpanFrame) {
   }
   return undefined;
 }
+
+// Looks at conventions described in https://graphql.org/learn/serving-over-http/
+// to infer if this is a GraphQL request, or GraphQL-like.
+// The SDK must be configured to capture the request body.
+export function isFrameGraphQLRequest(frame: SpanFrame): boolean {
+  if (!isRequestFrame(frame) || !frame.data?.request) {
+    return false;
+  }
+
+  const request = frame.data.request;
+  const contentType = request.headers?.['content-type'] ?? '';
+
+  // https://graphql.org/learn/serving-over-http/#headers
+  // `application/graphql-response+json` is a good indicator
+  if (contentType.includes('application/graphql-response+json')) {
+    return true;
+  }
+  // Legacy servers required `application/json`
+  if (!contentType.includes('application/json')) {
+    return false;
+  }
+
+  try {
+    const url = new URL(frame.description);
+    const hasUrlQuery = Boolean(url.searchParams.get('query'));
```

Comment:
Would this be a false positive for our API requests since a lot of them use `query`? What would it look like in that case?

Reply:
hrm, yeah it could. 

but it seems like we don't show any UI if we can't extract the names of things with `parseGraphQLQuery`. So looking at the preview site we're not showing graphql for requests on sentry.io. 

At minimum we could rename this function from `isFrameGraphQLRequest` to like `isMaybeGraphQLFrameRequest` or something like that.